app/models/profile_pin.rb

```diff
@@ -1,21 +1,40 @@
-class ProfilePin < ApplicationRecord
+class ProfilePin < ApplicationRecord 
   belongs_to :pinnable, polymorphic: true
   belongs_to :profile, polymorphic: true
+  belongs_to :article
 
   validates :profile_id, presence: true
-  validates :profile_type, inclusion: { in: %w[User] } # Future could be organization, tag, etc.
+  validates :profile_type, inclusion: { in: %w[User Article] } # Future could be organization, tag, etc.
   validates :pinnable_id, presence: true, uniqueness: { scope: %i[profile_id profile_type pinnable_type] }
-  validates :pinnable_type, inclusion: { in: %w[Article] } # Future could be comments, etc.
+  validates :pinnable_type, inclusion: { in: %w[Article Comment] } # Future could be comments, etc.
   validate :only_five_pins_per_profile, on: :create
+  validate :pin_comment_to_article
   validate :pinnable_belongs_to_profile
+  validate :article_to_article
 
   private
 
+  def get_profile_pins
+
+    #get all profile pins where an article has pinned comments
```

Comment:
**suggestion (non-blocking):** I don't think this and the next comment add too much, we could remove them.